C/services/common-plugin-interfaces/python/include/python_plugin_common_interface.h

```diff
@@ -0,0 +1,414 @@
+	PyObject* pReturn = PyObject_CallFunction(pFunc,
+						  "s",
+						   config->itemsToJSON().c_str());
+
+	Py_CLEAR(pFunc);
+
+	// Handle returned data
+	if (!pReturn)
+	{
+		Logger::getLogger()->error("Called python script method plugin_init : "
+					    "error while getting result object");
+		logErrorMessage();
+
+		PyGILState_Release(state);
+		return NULL;
+	}
+	else
+	{
+		Logger::getLogger()->info(">>>> OK plugin_handle: plugin_init(): "
+					  "got handle from python plugin='%p'",
+					  pReturn);
+
+		PyGILState_Release(state);
+		return (PLUGIN_HANDLE) pReturn;
+	}
+}
+
+/**
+ * Function to invoke 'plugin_reconfigure' function in python plugin
+ *
+ * @param    handle	The plugin handle from plugin_init_fn
+ * @param    config	The new configuration, as string
+ */
+static void plugin_reconfigure_fn(PLUGIN_HANDLE* handle,
+				  const std::string& config)
+{
+	std::mutex mtx;
+	PyObject* pFunc;
+	lock_guard<mutex> guard(mtx);
+	PyGILState_STATE state = PyGILState_Ensure();
+
+	//Logger::getLogger()->info("plugin_handle: plugin_reconfigure(): pModule=%p, *handle=%p", pModule, *handle);
+
+	// Fetch required method in loaded object
+	pFunc = PyObject_GetAttrString(pModule, "plugin_reconfigure");
+	if (!pModule || !pFunc)
+	{
+		Logger::getLogger()->info("plugin_handle: plugin_reconfigure(): pModule=%p, pFunc=%p",
+					   pModule,
+					   pFunc);
+	}
+
+	if (!pFunc || !PyCallable_Check(pFunc))
+	{
+		// Failure
+		if (PyErr_Occurred())
+		{
+			logErrorMessage();
+		}
+
+		Logger::getLogger()->fatal("Cannot find method plugin_reconfigure in loaded python module");
+		Py_CLEAR(pFunc);
+
+		PyGILState_Release(state);
+		return;
+	}
+
+	// Call Python method passing an object
+	PyObject* pReturn = PyObject_CallFunction(pFunc, "Os", *handle, config.c_str());
+
+	Py_CLEAR(pFunc);
+
+	// Handle returned data
+	if (!pReturn)
+	{
+		Logger::getLogger()->error("Called python script method plugin_reconfigure "
+					    ": error while getting result object");
+		logErrorMessage();
+		//*handle = NULL; // not sure if this should be treated as unrecoverable failure on python plugin side
+		return;
+	}
+	else
+	{
+		Py_CLEAR(*handle);
+		*handle = pReturn;
+		Logger::getLogger()->info("plugin_handle: plugin_reconfigure(): "
+					  "got updated handle from python plugin=%p",
+					  *handle);
+	}
+	PyGILState_Release(state);
+	return;
+}
+
+/**
+ * Function to log error message encountered while interfacing with
+ * Python runtime
+ */
+static void logErrorMessage()
+{
+	//Get error message
+	PyObject *pType, *pValue, *pTraceback;
+	PyErr_Fetch(&pType, &pValue, &pTraceback);
+	PyErr_NormalizeException(&pType, &pValue, &pTraceback);
+
+	PyObject* str_exc_value = PyObject_Repr(pValue);
+	PyObject* pyExcValueStr = PyUnicode_AsEncodedString(str_exc_value, "utf-8", "Error ~");
+	const char* pErrorMessage = pValue ?
+				    PyBytes_AsString(pyExcValueStr) :
+				    "no error description.";
+	Logger::getLogger()->fatal("logErrorMessage: Error '%s' ",
```

Comment:
This is logging Python errors? Can we make it clearer to the end user what this related to?

- Which Python module
- Which entry point